framework/src/st_things/things_stack/things_stack.c

```diff
@@ -385,11 +388,30 @@ int things_start_stack(void)
 		THINGS_LOG_E(TAG, "es_conn_type_Unknown");
 		return 0;
 	}
-
 	THINGS_LOG_V(TAG, "Stack Start Success");
 	return 1;
 }
 
+int things_start_scanning_ap() 
+{
+	pthread_mutex_unlock(&g_things_scan_ap_mutex);
+	if (pthread_create_rtos(&h_thread_things_scan_ap, NULL, auto_scanning_loop, NULL, THINGS_STACK_AP_SCAN_THREAD) != 0) {
+		THINGS_LOG_E(TAG, "Failed to create thread");
+		return 0;
+	}
+	return 1;
+}
+
+static void *__attribute__((optimize("O0"))) auto_scanning_loop(void)
+{
+	while (things_is_connected_ap() == false) {
```

Comment:
As Is: while (things_is_connected_ap() == false)
To Be: while (!things_is_connected_ap())